olap/src/aggregator/ceramic_patch.rs

```diff
@@ -60,19 +61,38 @@ impl WindowUDFImpl for CeramicPatch {
         Ok(Box::new(CeramicPatchEvaluator))
     }
 }
+// Small wrapper container around the data/state fields to hold
+// other mutable metadata for the event.
+// This is specific to Model Instance Documents.
+// Metadata is considered to be mutable from event to event and a overriting merge is performed
+// with the previous metadata to the current metadata.
+// This means if a metadata key is missing it is propogated forward until a new data event changes
+// its value.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct MIDDataContainer<D> {
+    metadata: BTreeMap<String, serde_json::Value>,
+    data: D,
+}
 
 #[derive(Debug)]
 struct CeramicPatchEvaluator;
 
 impl CeramicPatchEvaluator {
     fn apply_patch(patch: &str, previous_state: &str) -> Result<String> {
-        let patch: Vec<json_patch::PatchOperation> = serde_json::from_str(patch)
+        let patch: MIDDataContainer<Vec<PatchOperation>> = serde_json::from_str(patch)
```

Comment:
is this going to work given that in the events themselves the field is called `header` but in the state it's called `metadata`?

Reply:
It's a bit surprising to me to see that we're using the same type (`MIDDataContainer`) for both events and states.  If it works that's fine, just want to make sure we don't have any issues from overloading the type like that.

Reply:
Yeah sorry about the confusion. The field is always called `metadata` in the context that this struct is used.

The translation from raw event to conclusion event renames the field from header to metadata. (This was intentional as most fields inside the header end up as dimensions and only shouldIndex ends up in the metadata).

I see the confusion that a conclusion event data field has the same structure as an aggregated state data. I'll consider using two different (yet similar types) to make it clear.

Reply:
Decided the structures are the same so a good use for generics, with some type aliases to make it clear the context in how they should be used.

```rust
struct MIDDataContainer<D> {
    metadata: BTreeMap<String, serde_json::Value>,
    data: D,
}

type MIDDataContainerPatch = MIDDataContainer<Vec<PatchOperation>>;
type MIDDataContainerState = MIDDataContainer<serde_json::Value>;
```